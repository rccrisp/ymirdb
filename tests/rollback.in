set a 1 2 3
snapshot
set b a 4
snapshot
set c b 5
snapshot
set d c 6
snapshot
list snapshots
list entries
list keys
rollback 3
list snapshots
list entries
list keys
drop 4
set d c 7
snapshot
list snapshots
bye