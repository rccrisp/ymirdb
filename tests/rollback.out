> ok

> saved as snapshot 1

> ok

> saved as snapshot 2

> ok

> saved as snapshot 3

> ok

> saved as snapshot 4

> 4
3
2
1

> d [c 6]
c [b 5]
b [a 4]
a [1 2 3]

> d
c
b
a

> ok

> 3
2
1

> c [b 5]
b [a 4]
a [1 2 3]

> c
b
a

> no such snapshot

> ok

> saved as snapshot 5

> 5
3
2
1

> bye
